components/datepicker/stories/template.js

```diff
@@ -47,45 +61,61 @@ export const Template = ({
 			aria-required=${ifDefined(isRequired ? "true" : "false")}
 			aria-haspopup="dialog"
 		>
-			${[
-				TextField({
+			${TextField({
+				...globals,
+				size: "m",
+				isQuiet,
+				isDisabled,
+				isReadOnly: readOnly,
+				isInvalid: isRange ? "" : isInvalid,
+				customClasses: [`${rootClass}-textfield`],
+				customInputClasses: isRange ? [`${rootClass}-input`, `${rootClass}-startField`] : [`${rootClass}-input`],
+				placeholder: "Choose a date",
+				name: "field",
+				value: globals.selectedDay
+					? new Date(globals.selectedDay).toLocaleDateString(lang)
+					: undefined,
+				onclick: function () {
+					if (!isOpen) updateArgs({ isOpen: true });
```

Comment:
```suggestion
					updateArgs({ isOpen });
```

We probably want to toggle it on and off right?

Reply:
I didn't originally write this line but I think it might be intentional. This is controlling the open of the calendar when you click in the textfield and I could see it causing usability concerns to have the calendar close when you click on the text-field again.